https://github.com/Chia-Network/chia-blockchain/actions/runs/2988643149

This run had two 404s, several hung job timeouts, and one other 'regular' issue.

https://github.com/Chia-Network/chia-blockchain/runs/8177629311?check_suite_focus=true#step:16:180
```
_ ERROR at setup of test_farmer_get_harvester_plots_endpoints[get_harvester_plots_duplicates-filtering9-None-True-7] _
[gw0] linux -- Python 3.10.6 /home/runner/work/chia-blockchain/chia-blockchain/venv/bin/python
tests/core/test_farmer_harvester_rpc.py:78: in harvester_farmer_environment
    await time_out_assert(15, have_connections, True)
tests/core/test_farmer_harvester_rpc.py:76: in have_connections
    return len(await farmer_rpc_cl.get_connections()) > 0
chia/rpc/rpc_client.py:56: in get_connections
    response = await self.fetch("get_connections", request)
chia/rpc/rpc_client.py:46: in fetch
    response.raise_for_status()
venv/lib/python3.10/site-packages/aiohttp/client_reqrep.py:1004: in raise_for_status
    raise ClientResponseError(
E   aiohttp.client_exceptions.ClientResponseError: 404, message='Not Found', url=URL('https://localhost:43517/get_connections')
```

https://github.com/Chia-Network/chia-blockchain/runs/8177629536?check_suite_focus=true#step:16:180
```
_ test_farmer_get_harvester_plots_endpoints[get_harvester_plots_valid-filtering1-size-True-20] _
[gw2] linux -- Python 3.8.13 /home/runner/work/chia-blockchain/chia-blockchain/venv/bin/python
tests/core/test_farmer_harvester_rpc.py:432: in test_farmer_get_harvester_plots_endpoints
    harvester_plots = (await harvester_rpc_client.get_plots())["plots"]
chia/rpc/harvester_rpc_client.py:16: in get_plots
    return await self.fetch("get_plots", {})
chia/rpc/rpc_client.py:46: in fetch
    response.raise_for_status()
venv/lib/python3.8/site-packages/aiohttp/client_reqrep.py:1004: in raise_for_status
    raise ClientResponseError(
E   aiohttp.client_exceptions.ClientResponseError: 404, message='Not Found', url=URL('https://localhost:32981/get_plots')
```

--- a/chia/simulator/block_tools.py
+++ b/chia/simulator/block_tools.py
@@ -184,9 +184,9 @@ class BlockTools:
                 private_ca_crt_and_key=self.ssl_ca_cert_and_key_wrapper.collateral.cert_and_key,
                 node_certs_and_keys=self.ssl_nodes_certs_and_keys_wrapper.collateral.certs_and_keys,
             )
-        with lock_config(root_path=root_path, filename="config.yaml"):
-            path = config_path_for_filename(root_path=root_path, filename="config.yaml")
-            path.write_text(path.read_text().replace("localhost", "127.0.0.1"))
+            with lock_config(root_path=root_path, filename="config.yaml"):
+                path = config_path_for_filename(root_path=root_path, filename="config.yaml")
+                path.write_text(path.read_text().replace("localhost", "127.0.0.1"))
         self._config = load_config(self.root_path, "config.yaml")
         if automated_testing:
             if config_overrides is None:

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -64,7 +64,7 @@ def block_tools_fixture(get_keychain) -> BlockTools:
 # to run the tests, change the `self_hostname` fixture
 @pytest_asyncio.fixture(scope="session")
 def self_hostname():
-    return "localhost"
+    return "127.0.0.1"
 
 
 # NOTE:

